### PR TITLE
Added documentation for NF4 mode

### PR DESCRIPTION
**Pull request by ljaljushkin**

### CODE REVIEW

File: docs/compression_algorithms/CompressWeights.md [hunk 1]
```diff
@@ -6,17 +6,44 @@
...
 
-- Compress weights of linear layers and embeddings to int8
+- Compress weights to 8-bit integer data type.
 
 ```python
 from nncf import compress_weights
 compressed_model = compress_weights(model)
 ```
 
+- Compress internal weights to nf4 data type with group size = 128.
```
Comment (MaximProshin) [inline docs/compression_algorithms/CompressWeights.md:26]: Why do you use 'internal' here and below but not in int8 case? Please align it.

File: docs/compression_algorithms/CompressWeights.md [hunk 2]
```diff
@@ -6,17 +6,44 @@
...
+from nncf import CompressWeightsMode
+compressed_model = compress_weights(model, mode=CompressWeightsMode.NF4, group_size=64, ratio=0.9)
+```
+
 ##### Limitations
 
 - The algorithm is supported for OpenVINO and PyTorch models.
 - The compression applies in-place.
 - The compressed model is not trainable.
+- NF4 support is experimental - models quantized to nf4 should not be faster models quantized to 8-bit integer on client CPUs.
```
Comment (MaximProshin) [inline docs/compression_algorithms/CompressWeights.md:47]: I think it's confusing. I would remove it as we don't know when CPU will support it or at lease rephrase it saying that models with nf4 compressed weights should not be faster than models with int8 compressed weights on client CPUs.

File: docs/compression_algorithms/CompressWeights.md (hunk 1 above)
Comment (MaximProshin) [reply docs/compression_algorithms/CompressWeights.md:26]: I see, but I believe it's still confusing. Why do you want to explicitly use 'internal'? I think it's enough to explain that first and last layers are quantized to int8

File: docs/compression_algorithms/CompressWeights.md (hunk 2 above)
Comment (ljaljushkin) [reply docs/compression_algorithms/CompressWeights.md]: corrected

File: docs/compression_algorithms/CompressWeights.md (hunk 1 above)
Comment (ljaljushkin) [reply docs/compression_algorithms/CompressWeights.md]: Removed mentioning of internal weight to reduce confusion, explicitly say about first and last layers

**Review by ljaljushkin — COMMENTED** [review]
Thanks for review, @MaximProshin! Please take a look at the corrected document once more.

File: docs/compression_algorithms/CompressWeights.md [hunk 3]
```diff
@@ -6,17 +6,44 @@
...
 
+- Compress weights to nf4 data type with group size = 128, except first embedding and last linear layers - they are compressed to 8-bit integer data type.
+
+```python
+from nncf import compress_weights
+from nncf import CompressWeightsMode
+compressed_model = compress_weights(model, mode=CompressWeightsMode.NF4)
+```
+
+- Compress part of weights to nf4 with a different group size and the rest layers to 8-bit integer data type.
```
Comment (AlexKoff88) [inline docs/compression_algorithms/CompressWeights.md:35]:
```suggestion
- Compress weights of 90% of layers to nf4 with the group size 64, and the rest of layers to 8-bit integer data type.
```

**Review by AlexKoff88 — APPROVED** [review]

**Review by MaximProshin — APPROVED** [review]

**Pull request merged**